### PR TITLE
Task rung

### PR DESCRIPTION
**Pull request by vixhha**
input_taskrung

### CODE REVIEW

File: Project_FreeRTOS/main/task_rung.h [hunk 1]
```diff
@@ -6,21 +6,63 @@
...
       {
         // Thực hiện nhiệm vụ của input để đóng gói và gửi đi
+        if(abs(rung_value-rung_lastValue1)>=rung_noiseThreshold)
+        {
+          rung_count++;
+          // ngưỡng báo động
+          if(rung_value > ... && rung_count>=... )// ngưỡng giá trị báo động 
+          {
+            sw_send.payload = BUZZER_SW_ENABLE;
+            Serial.print("Buzzer on");
```
Comment (thaimq) [inline Project_FreeRTOS/main/task_rung.h:38]: Move line #28 to below this line, we shall update the payload before sending.

File: Project_FreeRTOS/main/task_rung.h [hunk 2]
```diff
@@ -6,21 +6,63 @@
...
+            rung_lastValue1=rung_value;
+
+          }
+          else 
+          {
+          sw_send.payload = BUZZER_SW_DISABLE;
+          Serial.print("Buzzer off");
```
Comment (thaimq) [inline Project_FreeRTOS/main/task_rung.h:45]: This one is the same above my comment, we should send the message to queue after updating the payload data

**Review by thaimq — CHANGES_REQUESTED** [review]
The logic code may difficult to understand, I think you should perform testing algorithm (create new project then write some sample code => deploy to arduino and check printing logs) before integration.

File: Project_FreeRTOS/main/task_rung.h [hunk 3]
```diff
@@ -6,21 +6,63 @@
...
+    unsigned long rung_currentTime = millis(); // đọc giá trị thời gian hiện tại
+
+
+    int rung_value = analogRead(IN_SW_1801P);
+    Serial.print("Rung Value: ");
+    Serial.println(rung_value); 
    Message sw_send = {0};
       sw_send.id_Tx = OUT_BUZZER;
       sw_send.id_Rx = IN_SW_1801P;
       sw_send.payload = BUZZER_SW_DISABLE;
```
Comment (ThanhtaDEV) [inline Project_FreeRTOS/main/task_rung.h:26]: Please swap the values ​​of these 2 lines 25&26
Tx = IN_SW_1801P;
Rx = OUT_BUZZER;

**Pull request closed without merging**